tests/test_objective_funs.py

```diff
@@ -569,6 +576,176 @@ def test(eq):
         test(get("HELIOTRON"))
 
 
+class TestQuadraticFlux:
+    """Tests quadratic flux."""
+
+    def get_grids(self, eq, field, N_grid=15, M_grid=10):
+        """Compute grids used for all of the tests."""
+        field_grid = LinearGrid(
+            rho=np.array([1.0]),
+            M=M_grid,
+            N=N_grid,
+            NFP=field.NFP if not isinstance(field, ToroidalMagneticField) else 1,
+            sym=False,
+        )
+        eval_grid = LinearGrid(
+            rho=np.array([1.0]),
+            M=eq.M_grid,
+            N=N_grid,
+            NFP=eq.NFP,
+            sym=False,
+        )
+
+        source_grid = LinearGrid(
+            rho=np.array([1.0]),
+            M=M_grid,
+            N=N_grid,
+            NFP=int(eq.NFP),
+            sym=False,
+        )
+
+        return field_grid, eval_grid, source_grid
+
+    def make_nonaxisym_field(self, eq):
+        """Makes a nonaxisymmetric field depending on the given eq."""
+        surf = eq.surface.copy()
+        surf.R_lmn = surf.R_lmn.at[surf.R_basis.get_idx(M=1, N=0)].set(2)
+        surf.Z_lmn = surf.Z_lmn.at[surf.Z_basis.get_idx(M=-1, N=0)].set(-2)
+        nonaxisym_field = FourierCurrentPotentialField.from_surface(
+            surface=surf,
+            Phi_mn=np.array([1e4]),
+            modes_Phi=np.array([[1, 0]]),
+            G=1e4,
+            I=0,
+            sym_Phi="sin",
+        )
+
+        return nonaxisym_field
+
+    @pytest.mark.unit
+    def test_quadratic_flux(self, quadratic_flux_equilibriums):
+        """Test calculation of quadratic flux on the boundary."""
+        t_field = ToroidalMagneticField(1, 1)
+
+        # test that torus (axisymmetric) Bnorm is exactly 0
+        __, eq = quadratic_flux_equilibriums
+        obj = QuadraticFlux(t_field, eq)
+        obj.build(eq, verbose=2)
+        f = obj.compute(field_params=t_field.params_dict)
+        np.testing.assert_allclose(f, 0, rtol=1e-15, atol=1e-15)
+
+        # test nonaxisymmetric surface
+        eq = desc.io.load("desc/examples/precise_QA_output.h5")[0]
```

Comment:
use `desc.examples.get` here, and see other comment about reducing resolution

Reply:
I didn't use `desc.examples.get` because this equilibrium worked better than the final equilibrium. Now that I think about it, it doesn't make much sense to me why the final equilibrium doesn't pass this test

Reply:
The "family" of the precise QA/QH are from the multigrid optimization, so each will have a different boundary shape, so the flux will be different.

you get get the first member using `get("precise_QA", "all")[0]` but we might also want to think if this is a good test for it since we don't expect the toroidal field to match well with it. Maybe @dpanici can make a coilset for this that will give low field error?

Reply:
was there a resolution to this?